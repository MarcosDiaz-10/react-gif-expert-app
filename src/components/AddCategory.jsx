import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {
  

    const [ inputValue, setInputValue ] = useState('');
  
    const onInputChange = ({ target}) => {
        setInputValue( target.value )
    }

    

    const onSubmit = ( event ) => {
        event.preventDefault();
        const clearCategory = inputValue.trim();

        if( clearCategory.length <= 1 ) return;

        onNewCategory( clearCategory );
        setInputValue('');

    }


    return (

        <form onSubmit={ onSubmit }>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue}
                onChange={ onInputChange }
            />
        </form>
  
    
    )
}

