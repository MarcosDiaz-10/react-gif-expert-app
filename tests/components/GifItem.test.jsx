import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas a GifItem', () => { 

    const title = 'Dragon Ball';
    const url = 'https://media3.giphy.com/media/IAi0BAedyQp6L0olPb/giphy.gif?cid=5ebce28cnrtk6zgvkwutk7l0ujyn1tejy09s2a6rlyuupufw&rid=giphy.gif&ct=g';    

    test('Debe hacer match con el snapshot', () => { 

        const { container } = render( <GifItem title={ title } url={ url }/>)

        expect( container ).toMatchSnapshot();
     })

     test('Debe de mostrar la imagen con el URL indicado y el ALT indicado', () => { 

        render( <GifItem title={ title } url={ url }/>);
        //   screen.debug();

        // expect ( screen.getByRole('img').src ).toBe( url )         
        // expect ( screen.getByRole('img').alt).toBe( title )         

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url);
        expect( alt ).toBe( title );
      })

      test('Debe de mostrar el titulo en el componente', () => { 

        render( <GifItem title={ title } url={ url }/>)

        expect( screen.getByText( title )).toBeTruthy();

       })
 })