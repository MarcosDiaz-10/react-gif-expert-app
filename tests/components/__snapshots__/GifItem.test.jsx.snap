// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://media3.giphy.com/media/IAi0BAedyQp6L0olPb/giphy.gif?cid=5ebce28cnrtk6zgvkwutk7l0ujyn1tejy09s2a6rlyuupufw&rid=giphy.gif&ct=g"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
